coins/getDBConnection.ts

```diff
@@ -0,0 +1,34 @@
+import postgres from "postgres";
+import setEnvSecrets from "./src/utils/shared/setEnvSecrets";
+
+let coins2Connection: ReturnType<typeof postgres>
+
+export async function getCoins2Connection() {
+  if (!coins2Connection) (coins2Connection as any) = 'isBeingSet'
+  if ((coins2Connection as any) === 'isBeingSet') {
```

Comment:
this looks like it will lead to race conditions

Reply:
if i call getCoins2Connection() twice when env is not yet set then i think it will create 2 connections, the first one will be overwritten and will just linger

Reply:
```
if (!coins2Connection) (coins2Connection as any) = new Promise(resolve=>{
...
resolve(postgres(auth[0]))
})
```
?

Reply:
you are right